lib/dw/dataset/applyChanges.js

```diff
@@ -1,39 +1,33 @@
 import _ from 'underscore';
 
-export default function(chart, dataset) {
-    var changes = chart.getMetadata('data.changes', []);
-    var transpose = chart.getMetadata('data.transpose', false);
-    // apply changes
+export default function applyChanges(chart, dataset) {
+    const changes = chart.getMetadata('data.changes', []);
+    const transpose = chart.getMetadata('data.transpose', false);
+    const numRows = dataset.numRows();
     changes.forEach(change => {
-        let row = 'row';
-        let column = 'column';
-        if (transpose) {
-            row = 'column';
-            column = 'row';
-        }
-
-        if (dataset.hasColumn(change[column])) {
+        const row = !transpose ? change.row : change.column;
+        const column = !transpose ? change.column : change.row;
+        if (dataset.hasColumn(column)) {
             change.ignored = false;
-            if (change[row] === 0) {
+            if (row === 0) {
+                // TODO: This should apply only when first row is header.
```

Comment:
@jakubdatawrapper I was referring to this line ^

Reply:
Ah yes, that. I didn't want to address that at the moment since it's not related to this PR. Nevertheless: Is the TODO correct? Is my understanding that the first row of the dataset doesn't always have to be the title correct?

Reply:
if the first row isn't the title we're automatically generating column names like "X.1", "X.2" etc. This would be done before `applyChanges()` so `row === 0` would still refer to the title. And in fact, users can overwrite these auto-generated titles just like they can overwrite uploaded column names.

Reply:
I see. Thanks for the insight! I've removed the TODO comment and will be merging the PR now.